### PR TITLE
Update the WebIDL parser

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/17254)

### CODE REVIEW

File: components/script/dom/webidls/Headers.webidl [hunk 1]
```diff
@@ -4,7 +4,7 @@
 
 // https://fetch.spec.whatwg.org/#headers-class
 
-typedef (Headers or sequence<sequence<ByteString>> or MozMap<ByteString>) HeadersInit;
+typedef (Headers or sequence<sequence<ByteString>> or record<DOMString, ByteString>) HeadersInit;
```
Comment (jdm) [inline components/script/dom/webidls/Headers.webidl]: The change from MozMap to record makes me uncomfortable; we definitely have not done any of the work from https://bugzilla.mozilla.org/show_bug.cgi?id=1330699 .

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 2]
```diff
@@ -1946,8 +1955,10 @@ def removeWrapperAndNullableTypes(types):
                 if parentName:
                     descriptor = descriptorProvider.getDescriptor(parentName)
                     extras += [descriptor.path, descriptor.bindingPath]
-            elif t.isType() and t.isMozMap():
+            elif t.isType() and t.isRecord():
                 extras += ['dom::bindings::mozmap::MozMap']
+            elif isinstance(t, IDLPromiseType):
```
Comment (jdm) [inline components/script/dom/bindings/codegen/CodegenRust.py]: t.isPromise()

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 3]
```diff
@@ -6870,6 +6883,8 @@ def process_arg(expr, arg):
             expr += ".r()"
         else:
             expr = "&" + expr
+    elif isinstance(arg.type, IDLPromiseType):
```
Comment (jdm) [inline components/script/dom/bindings/codegen/CodegenRust.py]: arg.type.isPromise()

**Review by jdm — DISMISSED** [review]

File: components/script/dom/bindings/codegen/CodegenRust.py (hunk 2 above)
Comment (nox) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Not all `t` have an `isPromise` method, probably an error but I don't want to have to maintain yet another `WebIDL.py` local patch for that.

**Pull request merged**